lib/active_model/serializer/associations.rb

```diff
@@ -61,26 +62,41 @@ def belongs_to(name, options = {})
         # @example
         #  has_one :author, serializer: AuthorSerializer
         #
-        def has_one(name, options = {})
-          associate HasOneReflection.new(name, options)
+        def has_one(name, options = {}, &block)
+          associate(HasOneReflection.new(name, options), &block)
         end
 
         private
 
-        # Add reflection and define {name} accessor.
+        # Add reflection and define {name} accessor and nested serializer.
         # @param [ActiveModel::Serializer::Reflection] reflection
         # @return [void]
         #
         # @api private
         #
-        def associate(reflection)
+        def associate(reflection, &block)
           self._reflections = _reflections.dup
 
           define_method reflection.name do
             object.send reflection.name
           end unless method_defined?(reflection.name)
 
-          self._reflections << reflection
+          _reflections << reflection
+
+          define_nested_serializer(reflection.name.to_s.singularize, &block) if block_given?
```

Comment:
I get that, i just mean design wise.. Also could check if block since you have the object

B mobile phone

> On Oct 2, 2015, at 8:13 AM, Lucas Hosseini <notifications@github.com> wrote:
> 
> In lib/active_model/serializer/associations.rb:
> 
> >            self._reflections = _reflections.dup
> >  
> >            define_method reflection.name do
> >              object.send reflection.name
> >            end unless method_defined?(reflection.name)
> >  
> > -          self._reflections << reflection
> > +          _reflections << reflection
> > +
> > +          define_nested_serializer(reflection.name.to_s.singularize, &block) if block_given?
> &block needs to be a param (i.e. named block) if you want to pass it as an argument to a method call. method(yield) does not work, nor does method do yield end (the self object gets messed up).
> 
> —
> Reply to this email directly or view it on GitHub.
>